app/src/main/java/de/westnordost/streetcomplete/quests/fire_hydrant_ref/AddFireHydrantRefForm.kt

```diff
@@ -0,0 +1,43 @@
+package de.westnordost.streetcomplete.quests.fire_hydrant_ref
+
+import android.os.Bundle
+import android.view.View
+import androidx.appcompat.app.AlertDialog
+import androidx.core.widget.doAfterTextChanged
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.databinding.QuestRefBinding
+import de.westnordost.streetcomplete.quests.AbstractOsmQuestForm
+import de.westnordost.streetcomplete.quests.AnswerItem
+import de.westnordost.streetcomplete.util.ktx.nonBlankTextOrNull
+
+class AddFireHydrantRefForm : AbstractOsmQuestForm<FireHydrantRefAnswer>() {
```

Comment:
Hm, as this is 100% exact the same as for postbox refs and emergency access point refs (and more?), I wonder if it is worth to have a generic `AddRefForm`?

Reply:
I think reducing code complexity/duplication is always worth it. Currently, I can't think of any necessary specialization. If we find some in the future, we can still undo this generalization.

Reply:
I agree. It's a simplification, and it's should be simple to revert if it turns out bad later.

Reply:
Yes, that's okay